lil_aretomo/aretomo.py

```diff
@@ -1,87 +1,76 @@
-        output_directory: Path,
-        aretomo_executable: Optional[Path] = None,
-        local_align: Optional[bool] = False,
-        target_pixel_size: Optional[float] = 10,
+        output_directory: PathLike,
+        expected_sample_thickness: int = 1500,
+        do_local_alignments: bool = False,
+        output_pixel_size: Optional[float] = 10,
         nominal_rotation_angle: Optional[float] = None,
-        n_patches_xy: Optional[Tuple[int, int]] = (5, 4),
-        thickness_for_alignment: Optional[float] = 1500,
-        correct_tilt_angle_offset: Optional[bool] = False,
-        gpu_ids: Optional[Tuple[int,...]] = None	
+        n_patches_xy: Optional[Tuple[int, int]] = None,
+        correct_tilt_angle_offset: bool = False,
+        gpu_ids: Optional[Sequence[int]] = None
 ):
-    """Run aretomo alignment on a single tilt-series
-    
+    """Align a single-axis tilt-series using AreTomo.
+
     Parameters
     ----------
-    
-    tilt_series: (n, y, x) array of tilt-images.
-    tilt_angles: nominal stage tilt-angles from the microscope.
-    pixel_size: pixel size of the tilt-series in angstroms-per-pixel
-    output_directory: tilt-series directory.
-    basename: basename for files in output directory.
-    (optional) aretomo_executable: path to the AreTomo executable file
-    (optional) local_align: carry out local tilt series alignments? Yes or no, default is no. 
-    (optional) target_pixel_size: the ideal pixel size at which TSA is carried out. Default is 10A
-    (optional) nominal_rotation_angle: initial estimate for the rotation angle of the tilt
-    axis. AreTomo does not need this information but it might help.
-    (optional) n_patches_xy: if local_align is True, AreTomo will carry out patch tracking. 
-    Specify the number of patches in X and Y here as tuple. Default is 5 in X,4 in Y.
-    (optional) thickness_for_alignment: thickness in Z in Ang for which AreTomo will use in the alignment.
-    This is useful is there is a lot of empty space at the top and bottom of your tomogram.
-    See AreTomo manual for full explanation: this sets -AlignZ. Default is 1500.
-    (optional) correct_tilt_angle_offset: Apply tilt angle offset correction, yes or no.
-    Default is no. See AreTomo manual for full explanation: yes; adds the -TiltCor 1 argument
-    (optional) gpu_ids: If you wish to run in parallel over multiple GPUs, enter GPU IDs as tuple e.g. 0 1 2 3
-    """
-    if check_aretomo_availability() is False:
-        e = 'AreTomo is not available for use. Load AreTomo so it can be called from the terminal via: AreTomo.'
-        console.log(f'ERROR: {e}')
-        raise RuntimeError(e)
+    tilt_series: (n, y, x) array of tilt images.
+    tilt_angles: nominal stage tilt angles.
+    pixel_size: 2D pixel spacing in Angstroms per pixel.
+    basename: basename for output files.
+    output_directory: directory for output files.
+    expected_sample_thickness
+    do_local_alignments
+    output_pixel_size
```

Comment:
this is separate from what's going on in RELION re: tomos so no stress there, good about the updates to tomo_model though!

We are now outputting a reconstruction rather than a TS here - will clarify docstring :)

Reply:
ooh so we have the option of using AreTomo for reconstruction (not just alignment) now? nice

Reply:
unfortunately not! this repo is a general purpose AreTomo wrapper, we are using this wrapper in RELION to do alignment in AreTomo but the same problems exist as before and we'll likely have to clean up the reconstructions